### PR TITLE
Split mixin file and add script to sync it

### PR DESCRIPTION
**Pull request by mcbenjemaa**
This PR:

- Split mixin file 
- add script to sync rules
- change the notes of mixin update

issue: https://github.com/giantswarm/giantswarm/issues/19855

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [X] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).
- [X] Alerting rules must have a comment documenting why it needs to exist.
- [X] Consider creating a dashboard (if it does not exist already) to help oncallers monitor the status of the issue.

### CODE REVIEW

**Comment by hervenicol** [comment]
Should we wait for https://github.com/giantswarm/dashboards/pull/124 to be merged before cleaning the doc?

**Comment by mcbenjemaa** [comment]
@hervenicol  Yes 👍

**Comment by mcbenjemaa** [comment]
@QuentinBisson We agreed to split the mixin rules file,

However, Do we need to add a script that syncs rules, the same as we did in dashboards?

**Comment by mcbenjemaa** [comment]
@QuentinBisson We agreed to split the mixin rules file,

However, Do we need to add a script that syncs rules, the same as we did in dashboards?

**Comment by QuentinBisson** [comment]
We will need to split the kube-prometheus-rules from the mixin files though

**Comment by mcbenjemaa** [comment]
@QuentinBisson  mixin rules are split and a script has been added to sync the rules.
can you take a look.

File: scripts/sync-kube-mixin.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# make a temporary dir to work in
+TMPDIR=$(mktemp -d -t 'tmp.XXXXXXXXXX')
```
Comment (hervenicol) [inline scripts/sync-kube-mixin.sh:8]: Missing quotes. Did you use `shellcheck`? ;)
There's a few other vars without quotes in the script.

File: scripts/sync-kube-mixin.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+trap 'rm -f $TMPDIR' EXIT
```
Comment (hervenicol) [inline scripts/sync-kube-mixin.sh:9]: Do you want to `rm -f` or `rm -rf` at exit, like at the end of the script?
Maybe that should be factorized in a common function?

File: scripts/sync-kube-mixin.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+MIXIN_REPO="git@github.com:giantswarm/giantswarm-kubernetes-mixin.git"
+# clone a branch of tag if provided
+BRANCH=${1:-""}
+
+if [ -z "${BRANCH}" ]; then
```
Comment (hervenicol) [inline scripts/sync-kube-mixin.sh:14]: I'm not a big fan of curly braces when not needed. I don't think it improves readability, it's just extra characters.
But it's my personal opinion / nitpicking ;)

File: scripts/sync-kube-mixin.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  name: kube-mixins.recording.rules
+  namespace: {{ .Values.namespace  }}
+spec:
+'
+
+# copy generated rules file
+cp "${TMPDIR}"/mixins/files/prometheus-rules/rules.yml ${RULESFILE}
+
+# prepend K8s objectmeta to the rules file
+printf '%s  %s' "${PRECONTENT}" "$(cat ${RULESFILE})" >${RULESFILE}
```
Comment (hervenicol) [inline scripts/sync-kube-mixin.sh:45]: Is this supposed to:
- write `$PRECONTENT`
- write `$RULESFILE` contents by adding an indentation at the beginning of each line?

If so, I don't understand how it works, and it does not seem to work on my laptop with bash.

On my laptop, here is what it does:
- write `$PRECONTENT`
- write `$RULESFILE` contents by adding an indentation at **the first** line

**Review by hervenicol — CHANGES_REQUESTED** [review]

File: scripts/sync-kube-mixin.sh (hunk 3 above)
Comment (mcbenjemaa) [reply scripts/sync-kube-mixin.sh]: I LIKE double braces in if conditions, I can distinguish that it's an if statement :D

File: scripts/sync-kube-mixin.sh (hunk 2 above)
Comment (hervenicol) [reply scripts/sync-kube-mixin.sh:9]: I mean, same code for same usage in 2 places, I thought it deserves a function. The syntax difference just proved the point :grin:

File: scripts/sync-kube-mixin.sh (hunk 3 above)
Comment (mcbenjemaa) [reply scripts/sync-kube-mixin.sh]: You suggest omitting curly braces `{ }` from each variable?

Comment (hervenicol) [reply scripts/sync-kube-mixin.sh]: Yes, when not needed. Just a suggestion, nothing critical there: we have no coding style guidelines for bash ;)

**Review by hervenicol — APPROVED** [review]
All good! :+1:

**Pull request merged**